### PR TITLE
EXAMPLEAND-29 fix null pointer exception

### PR DESCRIPTION
**Pull request by marcin-onegini**

### CODE REVIEW

File: app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java [hunk 1]
```diff
@@ -115,25 +116,36 @@ private boolean isBlank(final String string) {
...
-    if(cancelButton == null || cancelButton.getVisibility() == View.GONE){
+  private boolean isCancelButtonVisible() {
+    if (cancelButton == null || cancelButton.getVisibility() == View.GONE) {
       return false;
     }
     return true;
   }
 
-  protected void cancelRequest(){
+  protected void setCancelButtonVisibility(final int visibility) {
```
Comment (robertszuba) [inline app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java]: If we pass `VISIBLE`/`GONE` only (no `INVISIBLE`) than maybe it's easier to pass a boolean. Then we don't need to validate if passed value is correct.

File: app/src/main/java/com/onegini/mobile/exampleapp/view/activity/MobileAuthenticationFingerprintActivity.java [hunk 2]
```diff
@@ -36,7 +36,7 @@ protected void setupUi() {
 
   @Override
   protected void setCancelButtonVisibility() {
```
Comment (robertszuba) [inline app/src/main/java/com/onegini/mobile/exampleapp/view/activity/MobileAuthenticationFingerprintActivity.java]: This method doesn't `override` anymore and can be private. The same rule applies for other activities changed in this MR

File: app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java [hunk 3]
```diff
@@ -44,8 +45,8 @@
   @SuppressWarnings({ "unused", "WeakerAccess" })
   @Bind(R.id.authenticator_message)
   TextView authenticatorMessage;
-
-  @SuppressWarnings({"unused"})
+  @Nullable
```
Comment (robertszuba) [inline app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java]: Currently all layout activities have this button (i.e. push with fingerprint) even if it's hidden and non-used. Maybe we should add it for "simple push" too, to make the implementation similar?

File: app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java [hunk 4]
```diff
@@ -115,25 +116,36 @@ private boolean isBlank(final String string) {
...
+  private boolean isCancelButtonVisible() {
+    if (cancelButton == null || cancelButton.getVisibility() == View.GONE) {
       return false;
     }
     return true;
   }
 
-  protected void cancelRequest(){
+  protected void setCancelButtonVisibility(final int visibility) {
+    if (cancelButton == null) {
```
Comment (robertszuba) [inline app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java]: If we add the button to simple push view, then we don't need to check null case, but simply set the view to visible or gone.

File: app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java (hunk 3 above)
Comment (marcin-onegini) [inline app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java]: I would prefer to leave it as is, this way you won't have to add cancel button if you won't use it. I think, that we included cancel  button in views, which are using it.

Comment (marcin-onegini) [inline app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java]: But ok, it looks better.

Comment (robertszuba) [inline app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java]: I don't like adding this view when it's not used either, but it is already added to views like push with fingerprint or push with pin, while it's not used there. So we should remove it from classes that don't need it or add it to all classes - as long as it's consistent solution.

**Pull request merged**